Sources/RxMoya/RxMoyaProvider.swift

```diff
@@ -21,7 +21,10 @@ open class RxMoyaProvider<Target>: MoyaProvider<Target> where Target: TargetType
 
         // Creates an observable that starts a request each time it's subscribed to.
         return Observable.create { [weak self] observer in
```

Comment:
That second method would mean we'd need to take all the initializer args to `RxMoyaProvider` in the class method. OR maybe we make this an instance method too of a different name?

I suppose both of these things can be left as exercises for the user though. Do we know what happens to an inflight request if a regular `MoyaProvider` is deallocated?